source/Ship.cpp

```diff
@@ -2361,7 +2377,8 @@ shared_ptr<Ship> Ship::Board(bool autoPlunder)
 		return shared_ptr<Ship>();
 
 	// For a fighter or drone, "board" means "return to ship."
-	if(CanBeCarried())
+	// Except when drone or fighter is boarding another drone or fighter.
+	if(victim->CanCarry(*this) && GetParent() == victim)
```

Comment:
@quyykk @tehhowch regarding your comments noted here https://github.com/endless-sky/endless-sky/pull/6726/files#r872249830

This is partially because fighter boarding logic is counter intuitive.  Refer to this block.  Carriers do not drive to and "pick up ships".  The opposite occurs.  Fighters always board the parent to be carried.  If a fighter is disabled it cannot board the parent i.e. cannot be carried.  This is how it works in vanilla logic because fighters visit carriers and not the other way around.  Heck, in vanilla carriers don't even board disabled fighters (probably because originally thought too complex to implement?).  This PR also fixes carriers boarding fighters to fix them.

It is this block specifically why fighters must not be disabled when they are near the parent.  In vanilla, it is impossible for fighters to be disabled by energy so there's never a scenario you need to worry about re-enabling when near the parent.  They are always enabled floating off into infinity.